app/controllers/backgrounds_controller.rb

```diff
@@ -42,20 +42,15 @@ def create
   # PATCH/PUT /backgrounds/1
   # PATCH/PUT /backgrounds/1.json
   def update
-    current_data = @background.to_json(include: [{ pipeline_runs: { only: [:id, :sample_id] } }])
-    current_data_full_string = @background.to_json(include: [{ pipeline_runs: { only: [:id, :sample_id] } },
-                                                             :taxon_summaries])
-    background_changed = assign_attributes_and_has_changed?(background_params)
-    if background_changed
-      s3_archive_successful, s3_backup_path = archive_background_to_s3(current_data_full_string)
-      db_archive_successful = archive_background_to_db(current_data, s3_backup_path)
-      update_successful = db_archive_successful && s3_archive_successful ? @background.save : false # this triggers recomputation of @background's taxon_summaries if archiving was successful
+    if background_params[:pipeline_run_ids].map(&:to_i).select { |p| p > 0 } .sort == @background.pipeline_runs.pluck(:id).sort
```

Comment:
i'm confused about the equality here -- is this to allow a background to be updated only if the pipeline_run ids exactly match? and then the only thing that is updated is the name? so this allows rename functionality, is that right?

Reply:
if that's the case, a comment would really help

Reply:
yes. but actual logic is more complicated. I am still working on it. will add comments.